src/ai/backend/manager/models/vfolder.py

```diff
@@ -1245,52 +1255,34 @@ async def initiate_vfolder_deletion(
         db_engine, vfolder_ids, VFolderOperationStatus.DELETE_ONGOING, do_log=False
     )
 
-    row_deletion_infos: list[VFolderDeletionInfo] = []
-    failed_deletion: list[tuple[VFolderDeletionInfo, str]] = []
-
-    async def _delete():
-        for vfolder_info in requested_vfolders:
-            folder_id, host_name = vfolder_info
-            proxy_name, volume_name = storage_manager.split_host(host_name)
-            try:
-                async with storage_manager.request(
-                    proxy_name,
-                    "POST",
-                    "folder/delete",
-                    json={
-                        "volume": volume_name,
-                        "vfid": str(folder_id),
-                    },
-                ) as (_, resp):
-                    pass
-            except (VFolderOperationFailed, InvalidAPIParameters) as e:
-                if e.status == 404:
-                    row_deletion_infos.append(vfolder_info)
-                else:
-                    failed_deletion.append((vfolder_info, repr(e)))
-            except Exception as e:
-                failed_deletion.append((vfolder_info, repr(e)))
-            else:
-                row_deletion_infos.append(vfolder_info)
-        if row_deletion_infos:
-            vfolder_ids = tuple(vf_id.folder_id for vf_id, _ in row_deletion_infos)
+    already_deleted: list[VFolderDeletionInfo] = []
 
-            await update_vfolder_status(
-                db_engine, vfolder_ids, VFolderOperationStatus.DELETE_COMPLETE, do_log=False
-            )
-            log.debug("Successfully removed vfolders {}", [str(x) for x in vfolder_ids])
-        if failed_deletion:
-            await update_vfolder_status(
-                db_engine,
-                [vfid.vfolder_id.folder_id for vfid, _ in failed_deletion],
-                VFolderOperationStatus.DELETE_ERROR,
-                do_log=False,
-            )
-            extra_data = {str(vfid.vfolder_id): err_msg for vfid, err_msg in failed_deletion}
-            raise VFolderOperationFailed(extra_data=extra_data)
+    for vfolder_info in requested_vfolders:
+        folder_id, host_name = vfolder_info
+        proxy_name, volume_name = storage_manager.split_host(host_name)
+        try:
+            async with storage_manager.request(
+                proxy_name,
+                "POST",
+                "folder/delete",
+                json={
+                    "volume": volume_name,
+                    "vfid": str(folder_id),
+                },
+            ) as (_, resp):
+                pass
+        except (VFolderOperationFailed, InvalidAPIParameters) as e:
+            if e.status == 404:
+                already_deleted.append(vfolder_info)
+    if already_deleted:
+        vfolder_ids = tuple(vf_id.folder_id for vf_id, _ in already_deleted)
```

Comment:
Does the 404 mean it's already been deleted?
It looks like a status code that is used even when the value is not present in the first place.

Reply:
do you mean it looks like 404 should be used when there is no DB record?

Reply:
I meant that it seems to return `404` even when there is no DB record, so it seems awkward to use it as a semantics for `already_deleted`.

Reply:
storage-proxy returns the 404 response when it does not find the folder, so it is either already deleted or not found. Could you suggest a better name for the variable `already_deleted`?

Reply:
I think it would be better to use the 410 gone status with it.
Can the storage-proxy now distinguish whether the requested value was deleted or never existed in the first place?
If the current structure makes it difficult, there's nothing wrong with going with the 404.

Reply:
This is a delete API and manager calls this.
As manager can request deletion of existing vfolders, it would be better to response with 410.